### PR TITLE
TMA-1190: Add test env clean-up script

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

File: bin/test_projects_cleanup.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      project_add && project_add.output_stage && project_add.output_stage.delete
+      project.delete
+    else
+      puts "Would delete: #{project.pid} - #{project.title} - #{project.created}"
+    end
+  end
+  puts "#{filtered_projects.length} projects matching \"#{title}\" #{'would be ' unless force}deleted."
+end
+
+def delete_ads_by_title(title, client, days = 14, force = false)
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has boolean parameter 'force' (BooleanParameter) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has 4 parameters (LongParameterList) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has approx 16 statements (TooManyStatements) (reek)

File: bin/test_projects_cleanup.rb [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  exit 0
+end
+
+require 'gooddata'
+require_relative '../spec/environment/environment'
+GoodData::Environment.load
+config = GoodData::Environment::ConnectionHelper::LCM_ENVIRONMENT
+secrets = GoodData::Environment::ConnectionHelper::SECRETS
+
+def delete_project_by_title(title, projects, days = 14, force = false)
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has boolean parameter 'force' (BooleanParameter) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has 4 parameters (LongParameterList) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has approx 10 statements (TooManyStatements) (reek)

File: bin/test_projects_cleanup.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  warehouses.each do |warehouse|
+    warehouse_title = warehouse.title
+    next unless warehouse_title.match(title)
+
+    dead_line = Time.now - days * 60 * 60 * 24
+    created = Time.parse(warehouse.data["created"])
+    next if created > dead_line
+
+    begin
+      if force
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Is controlled by argument 'force' (ControlParameter) (reek)

File: bin/test_projects_cleanup.rb [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+config = GoodData::Environment::ConnectionHelper::LCM_ENVIRONMENT
+secrets = GoodData::Environment::ConnectionHelper::SECRETS
+
+def delete_project_by_title(title, projects, days = 14, force = false)
+  dead_line = Time.now - days * 60 * 60 * 24
+  filtered_projects = projects.select do |p|
+    p.title.match(title) && p.created < dead_line
+  end
+  filtered_projects.each do |project|
+    if force
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Is controlled by argument 'force' (ControlParameter) (reek)

File: bin/test_projects_cleanup.rb [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      puts "Deleting: #{project.pid} - #{project.title} - #{project.created}"
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Calls 'project.created' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Calls 'project.pid' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Calls 'project.title' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Refers to 'project' more than self (maybe move it to another class?) (FeatureEnvy) (reek)

File: bin/test_projects_cleanup.rb [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+      project_add = project.add
+      project_add && project_add.output_stage && project_add.output_stage.delete
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Calls 'project_add.output_stage' 2 times (DuplicateMethodCall) (reek)

File: bin/test_projects_cleanup.rb [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+      end
+      deleted += 1
+    rescue StandardError => e
+      puts "Failed to delete #{warehouse_title}: #{e}"
+    end
+  end
+  puts "#{deleted} ADS instances with title \"#{title}\" #{'would be ' unless force}deleted."
+end
+
+def clean_up!(client, force, days)
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has approx 13 statements (TooManyStatements) (reek)

File: bin/test_projects_cleanup.rb [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+    begin
+      if force
+        puts "Deleting: #{warehouse_title} - #{created}"
+        warehouse.delete
+      else
+        puts "Would delete:  #{warehouse_title} - #{created}"
+      end
+      deleted += 1
+    rescue StandardError => e
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has the variable name 'e' (UncommunicativeVariableName) (reek)

File: bin/test_projects_cleanup.rb [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+require 'gooddata'
+require_relative '../spec/environment/environment'
+GoodData::Environment.load
+config = GoodData::Environment::ConnectionHelper::LCM_ENVIRONMENT
+secrets = GoodData::Environment::ConnectionHelper::SECRETS
+
+def delete_project_by_title(title, projects, days = 14, force = false)
+  dead_line = Time.now - days * 60 * 60 * 24
+  filtered_projects = projects.select do |p|
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Has the variable name 'p' (UncommunicativeVariableName) (reek)

File: bin/test_projects_cleanup.rb [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+  delete_project_by_title(/lcm-test-fixture/, projects, days, force)
+  delete_project_by_title(/Test MASTER project/, projects, days, force)
+  delete_project_by_title(/Test MINOR project/, projects, days, force)
+  delete_project_by_title(/^Test project$/, projects, days, force)
+  delete_project_by_title(/userprov-e2e-testing/, projects, days, force)
+  delete_ads_by_title(/Development ADS/, client, days, force)
+  delete_ads_by_title(/Production ADS/, client, days, force)
+end
+
+def init_client(username, password, server)
```
Comment (yenkins) [inline bin/test_projects_cleanup.rb]: info: Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction) (reek)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/858/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1671/ : SUCCESS in 2m 53s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1932/ : SUCCESS in 2m 43s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/109/ : SUCCESS in 9m 16s

**Comment by panjan** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/860/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1673/ : SUCCESS in 1m 04s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1934/ : FAILURE in 27s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/111/ : SUCCESS in 6m 54s

**Comment by panjan** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/861/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1674/ : SUCCESS in 1m 04s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1935/ : FAILURE in 19s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/112/ : SUCCESS in 7m 35s

**Comment by panjan** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/862/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1675/ : FAILURE in 19s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1936/ : SUCCESS in 2m 32s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/113/ : SUCCESS in 9m 37s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/864/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1677/ : SUCCESS in 1m 09s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1938/ : SUCCESS in 1m 11s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/115/ : SUCCESS in 7m 40s

**Comment by panjan** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/865/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1678/ : SUCCESS in 2m 51s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1939/ : SUCCESS in 1m 11s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/116/ : SUCCESS in 7m 59s

**Comment by gdgate** [comment]
Build in pipeline gate aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/135/ : SUCCESS in 13s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-pipeline/66/ : FAILURE in 14s

**Comment by Hahihula** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/136/ : SUCCESS in 15s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-pipeline/67/ : FAILURE in 12s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/137/ : SUCCESS in 18s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-pipeline/68/ : FAILURE in 14s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/138/ : SUCCESS in 17s

**Pull request merged**